src/components/PersonLink.tsx

```diff
@@ -0,0 +1,37 @@
+import classNames from 'classnames';
+import { Person } from '../types';
+import { useNavigate } from 'react-router-dom';
+
+type PeopleLinkProps = {
+  person: Person | null;
+  searchParams: string;
+};
+
+export const PersonLink: React.FC<PeopleLinkProps> = ({
+  person,
+  searchParams,
+}) => {
+  const navigate = useNavigate();
+
+  if (!person) {
+    return <>{'-'}</>;
+  }
+
+  const handleNavigateWithParams = (
+    e: React.MouseEvent<HTMLAnchorElement>,
+    path: string,
+  ) => {
+    e.preventDefault();
+    navigate(`${path}${location.search}`);
```

Comment:
The `location` object is not imported or defined in this component. You should use `searchParams` directly instead of `location.search` to ensure the correct query string is appended to the navigation path.